### PR TITLE
Add similar-note feature

### LINKED ISSUE #25: Note suggestions and aliases
If you don't get the `!note` call right, we should show a suggestion like "Hey, did you mean .ddragon?"

I'm not sure how to approach this. Could use some expertise :)

### PR DESCRIPTION
**Pull request by stelar7**
fixes #25

### CODE REVIEW

**Comment by Querijn** [comment]
nice

File: src/Info.ts [hunk 1]
```diff
@@ -162,11 +162,34 @@ export default class Info {
...
+            });
+
+            let message = "Did you mean: ";
+            let found = false;
+            similarity.forEach(s => {
+                if (s.score < 2) {
+                    message += "`" + s.command + "`, ";
+                    found = true;
+                }
+            });
```
Comment (Querijn) [inline src/Info.ts]: This could be a map() and join()'d, would save that last slice

File: src/Info.ts [hunk 2]
```diff
@@ -176,4 +199,40 @@ export default class Info {
...
+        if (a.length === 0) {
+            return b.length;
+        }
+
+        if (b.length === 0) {
+            return a.length;
+        }
+
+        let v0 = [];
+        const v1 = [];
```
Comment (Querijn) [inline src/Info.ts]: one const, one let?

Comment (stelar7) [reply src/Info.ts]: v0 is changed later in the loop, so it cant be const.

File: src/Info.ts [hunk 3]
```diff
@@ -162,11 +162,34 @@ export default class Info {
...
+            let found = false;
+            similarity.forEach(s => {
+                if (s.score < 2) {
+                    message += "`" + s.command + "`, ";
+                    found = true;
+                }
+            });
+
+            if (found) {
+                return { message: message.slice(0, -2) + "?", counter: 0 } as InfoData;
```
Comment (molenzwiebel) [inline src/Info.ts]: Do you really need to cast here?

File: src/Info.ts [hunk 4]
```diff
@@ -162,11 +162,34 @@ export default class Info {
...
+            }
+
+            return { message: `No note with the name ${command} was found.`, counter: 0 } as InfoData;
```
Comment (molenzwiebel) [inline src/Info.ts]: ^

File: src/Info.ts [hunk 5]
```diff
@@ -176,4 +199,40 @@ export default class Info {
         info.counter++;
         return info;
     }
+
+    private levenshteinDistance(a: string, b: string): number {
```
Comment (molenzwiebel) [inline src/Info.ts]: Cmon stelar, this is not the node mentality. Get yourself a package instead! [fast-levenshtein](https://www.npmjs.com/package/fast-levenshtein) has typescript typings available at `@types/fast-levenshtein`.

File: src/Info.ts (hunk 3 above)
Comment (stelar7) [reply src/Info.ts]: Yeah

File: src/Info.ts (hunk 4 above)
Comment (stelar7) [reply src/Info.ts]: ^

File: src/Info.ts (hunk 3 above)
Comment (Querijn) [reply src/Info.ts]: If you require a cast, it means something isn't set. In this case, you're missing "command"

**Pull request merged**